core/src/main/java/io/undertow/server/handlers/resource/ResourceHandler.java

```diff
@@ -159,7 +159,13 @@ public void handleRequest(final HttpServerExchange exchange) throws Exception {
             } else {
                 exchange.setStatusCode(StatusCodes.NOT_IMPLEMENTED);
             }
-            exchange.endExchange();
+
+            if ( this.next != null ) {
```

Comment:
This would only get hit if the method is none of the following.

```java
        KNOWN_METHODS.add(Methods.OPTIONS);
        KNOWN_METHODS.add(Methods.GET);
        KNOWN_METHODS.add(Methods.HEAD);
        KNOWN_METHODS.add(Methods.POST);
        KNOWN_METHODS.add(Methods.PUT);
        KNOWN_METHODS.add(Methods.DELETE);
        KNOWN_METHODS.add(Methods.TRACE);
        KNOWN_METHODS.add(Methods.CONNECT);
```

They all get handled above this and will return the method not allowed.

Your code would only trigger if it is a non standard HTTP method. It appears it will also still set the status code to `NOT_IMPLEMENTED` which your handler may or may not be expecting. It will also break the normal functionality of this handler which someone might rely on. It might be better to make this an opt in functionality rather than changing the default behavior.

Reply:
Do you want your logic to replace the inside of https://github.com/undertow-io/undertow/pull/656/files#diff-8c222f88bc42bde6bcc77cb5a15dbb08R155 instead? Then you could add a new handler that defaults to the old functionality and allow someone to override it if they wish. `resourceHandler.setUnsupportedMethodHandler()`

Reply:
I agree. 

So this is the action list:

1. Add a new field of type HttpHandler, called unsupportedMethodHandler; Move the else clause (Line#155) inside of it.
2. A setter ( which verifies the argument isn't null )